src/Build.OM.UnitTests/Definition/ProjectItem_Tests.cs

```diff
@@ -2158,32 +2158,64 @@ public void RemoveWithItemReferenceOnMatchingMetadata()
                 <I2 Include='c2' M1='3' m2='Y'/>
                 <I2 Include='d2' M1='y' m2='d'/>
 
-                <I2 Remove='@(I1)' MatchOnMetadata='M1;m2'/>");
+                <I2 Remove='@(I1)' MatchOnMetadata='M1'/>");
 
             var project = ObjectModelHelpers.CreateInMemoryProject(content);
             var items = project.ItemsIgnoringCondition.Where(i => i.ItemType.Equals("I2"));
 
-            items.Select(i => i.EvaluatedInclude).ShouldBe(new[] { "a2", "c2", "d2" });
+            items.Select(i => i.EvaluatedInclude).ShouldBe(new[] { "a2", "d2" });
 
             items.ElementAt(0).GetMetadataValue("M1").ShouldBe("x");
             items.ElementAt(0).GetMetadataValue("M2").ShouldBe("c");
-            items.ElementAt(1).GetMetadataValue("M1").ShouldBe("3");
-            items.ElementAt(1).GetMetadataValue("M2").ShouldBe("Y");
-            items.ElementAt(2).GetMetadataValue("M1").ShouldBe("y");
-            items.ElementAt(2).GetMetadataValue("M2").ShouldBe("d");
+            items.ElementAt(1).GetMetadataValue("M1").ShouldBe("y");
+            items.ElementAt(1).GetMetadataValue("M2").ShouldBe("d");
         }
 
         [Fact]
-        public void RemoveWithItemReferenceOnMatchingMetadataReferences()
+        public void RemoveWithItemReferenceOnMatchingMetadataPropertyReferences()
```

Comment:
nit: simpler name for this and next test would be `RemoveWithPropertyReferenceInMatchOnMetadata` and `RemoveWithItemReferenceInMatchOnMetadata`